### PR TITLE
Express State Management Options in README

### PR DESCRIPTION
**Pull request by thomascothran**
Problem
-------
Visitors to Zero's repository would benefit from seeing very basic examples of building stateful components, so that they can see what using Zero looks like.

Solution
--------
Use the simple example of an incrementing button to show the different options for managing state.

### CODE REVIEW

**Comment by thomascothran** [comment]
This is still in draft. I want to discuss the approach before filling out the other examples.

**Comment by thomascothran** [comment]
Initially I was thinking to include more examples. However, I think these three are enough to give visitors a feel for Zero, using the DOM, reagent, and re-frame -esque examples.

Let me know what you think @raystubbs

File: README.md [hunk 1]
```diff
@@ -29,6 +29,111 @@ A toolkit for building web components in Clojure and ClojureScript.
...
+  [:root> {::z/on {:click (z/act [::db/patch [{:path [:clicks]
+                                               :fn inc}]])}}
+   [:button (str "Clicked " clicks " times")]])
+
+(zc/reg-components
+ :incrementing-button {:view button-view
+                       :props {:clicks (z/bnd ::db/path [:clicks])}})
+```
+
+Here we have an in-memory database for our application. We bind our clicks prop to a path in the database, and then use the `::db/path` effect to update the value at the `:clicks` path.
```
Comment (raystubbs) [inline README.md]: Typo `::db/path -> ::db/patch`.

**Review by raystubbs — DISMISSED** [review]
Looks great!  Thanks for all the work.

File: README.md (hunk 1 above)
Comment (thomascothran) [reply README.md]: Good catch! Fixed.

**Pull request merged**